clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -1069,7 +1069,7 @@ void Clang::AddPreprocessingOptions(Compilation &C, const JobAction &JA,
 
       // If user provided -o, that is the dependency target, except
       // when we are only generating a dependency file.
-      Arg *OutputOpt = Args.getLastArg(options::OPT_o);
+      Arg *OutputOpt = Args.getLastArg(options::OPT_o, options::OPT__SLASH_Fo);
```

Comment:
This change doesn't affect the test. Let's either add a test that exercises this, or drop this change.

Reply:
Added a test.